### ISSUE TITLE
istio-cni: use/support CSI driver for CNI <-> zt shared mount to avoid hostmounts

### ISSUE THREAD

**Issue by bleggett** [istio/istio#51468]
**Describe the feature request**

Similar to what SPIRE does, we should use (or support the use of) a CSI driver to manage the shared socket mount volume between CNI and ztunnel. This allows both CNI and ztunnel to have reduced permissions, and enables us to work in environments where pods are forbidden from creating raw hostmounts.

**Affected product area (please put an X in all that apply)**

[x] Ambient
[ ] Docs
[ ] Dual Stack
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by howardjohn** [comment]
Doesn't CNI need hostmount no matter what? So we can only really reduce the ztunnel one?

CNI could nsenter into ztunnel's namespace and use an anonymous UDS as well perhaps

**Comment by bleggett** [comment]
> Doesn't CNI need hostmount no matter what? So we can only really reduce the ztunnel one?

Yes this issue is in reference to the ZDS socket, which is a privileged communication socket that we create on the host and share among multiple pods, which is exactly what CSI is designed to handle.

Much like we do with spire we can avoid a UDS host mount.

> CNI could nsenter into ztunnel's namespace and use an anonymous UDS as well perhaps

It could, but the CSI driver is the way this is typically handled in K8S, and may already be in use anyway if things like SPIRE are present.

And I don't really want to have two ways to do this, there's no need for that that I can see.

**Comment by howardjohn** [comment]
CSI means one of two t hings to me:

1) implement the CSI spec in CNI
2) Use an off-the-shelf CSI

Were you thinking one of those or something else?

(1) is incredibly complex in my (very limited!) experience. (2) seems overkill to have 3 daemonsets on the node

**Comment by bleggett** [comment]
As an alternative to raw hostmounts, since there are some envs where raw hostmounts are frowned upon - whichever is easiest.

I think all we need for (1) is a CSI driver, which based on [the SPIRE example](https://github.com/spiffe/spiffe-csi) doesn't look that bad.

Whether it should be its own thing, part of `istio-cni` or part of `istiod` is very much TBD.

Given that it doesn't look too bad and CSI drivers are a pretty normative way to solve this problem in K8S from what I can tell, I have an initial preference for that versus doing more Istio-specific NS tricks in `istio-cni` - I suspect doing that would amount to the same amount of code and be harder to explain to people versus the CSI driver route.

This is mostly an investigative ticket - if someone asks about X, we can investigate doing it via Y, which _will_ work, though we could come up with something better (Z). It's not strictly pressing to solve this now tho.
